chart/monitoring/index.vue

```diff
@@ -80,16 +99,117 @@ export default {
   },
 
   methods: {
+    async createNamespace(cb) {
+      const { value: model, targetNamespace } = this;
+
+      if (model.alertmanager.enabled) {
+        if (targetNamespace) {
+          return targetNamespace;
+        }
+
+        try {
+          const nsData = { type: NAMESPACE, metadata: { name: this.chart.targetNamespace } };
+
+          const neuNS = await this.$store.dispatch('cluster/create', nsData);
+
+          await neuNS.save();
+          await neuNS.waitForState('active');
+
+          return neuNS;
+        } catch (error) {
+          console.error(error); // eslint-disable-line no-console
+          throw error;
+        }
+      }
+    },
+
+    async createDefaultSecret(cb) {
+      const { value: model, newAlertManagerSecret } = this;
+
+      if (model.alertmanager.enabled && !model.alertmanager.alertmanagerSpec.useExistingSecret) {
+        if (isEmpty(newAlertManagerSecret)) {
+          throw new Error('An AlertManager secret is required');
+        }
+
+        if (!has(newAlertManagerSecret?.data, 'alertmanager.yaml')) {
+          throw new Error('The AlertManager config must have a key with alertmanager.yaml');
+        }
+
+        try {
+          await newAlertManagerSecret.save();
+          await newAlertManagerSecret.waitForState('active');
+
+          return newAlertManagerSecret;
+        } catch (error) {
+          console.error(error); // eslint-disable-line no-console
+          throw error;
+        }
+      }
+    },
+
     async fetchDeps() {
-      const storageClasses = await this.$store.dispatch('cluster/findAll', { type: STORAGE_CLASS });
-      const pvcs = await this.$store.dispatch('cluster/findAll', { type: PVC });
+      const { $store } = this;
+      const storageClasses = await $store.dispatch('cluster/findAll', { type: STORAGE_CLASS });
+      const pvcs = await $store.dispatch('cluster/findAll', { type: PVC });
+      const secrets = await $store.dispatch('cluster/findAll', { type: SECRET });
```

Comment:
Fetch these in parallel with all/allHash instead of waiting for each other

Reply:
updated this block